cmd/clusterctl/client/cluster/crd_migration.go

```diff
@@ -42,6 +42,11 @@ type crdMigrator struct {
 	Client client.Client
 }
 
+// NewCRDMigrator public function to create a new CRD migrator.
+func NewCRDMigrator(client client.Client) *crdMigrator { //nolint:revive
```

Comment:
Should this return an interface instead of an unexported type?

Usually in go public funcs shouldn't return non-exported types